### ISSUE TITLE
pulseeffects and cpu-usage

### ISSUE THREAD

**Issue by Rosika2** [wwmm/pulseeffects#939]
not a bug, rather a question

Hi all,

my system is **Linux/Lubuntu 20.04.2 LTS** and I´m using **pulseeffects**. 
It works nicely and generally I´m very satisifed with it. I have on question though:

When running pulseeffects (with two tabs active: "automatic gain conrol" and "equalizer") cpu-usage is definitively higher:

"top -i" lists it as:

```
PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     ZEIT+ BEFEHL                   
6242 rosika    20   0 1709964  96032  35952 S  17,6   2,4   0:41.38 pulseeffects   
```

my cpu-specs (using inxi) are thus:
          
```
inxi -Fz
    [...]
    CPU:       Topology: Dual Core model: Intel Core i3-3240T bits: 64 type: MT MCP 
               L2 cache: 3072 KiB 
               Speed: 1596 MHz min/max: 1600/2900 MHz Core speeds (MHz): 1: 1596 2: 1596 3: 1596 
               4: 1596 
```
               
My question is: Is this behaviour to be expected?   

Thanks a lot in advance.

Many greetings.
Rosika

**Comment by wwmm** [comment]
>with two tabs active: "automatic gain control" and "equalizer"

The autogain plugin is heavy on resources. It is based on perceived loudness estimations done by the library `libebur128` and the calls to this library functions are expensive. 

Depending on which pulseeffects version you are using something you can try is increasing the latency or the buffer size in our settings menu. Increasing the latency reduces the number of calls to this library. As a result less cpu will be used.

**Comment by Rosika2** [comment]
@wwmm:

Hi and thank you so much for your answer.

> which pulseeffects version you are using

I looked it up and its version 4.7.1.

I´ll try experimenting with **latency** and **buffer size**. Thanks for the suggestion.

Applying the new knowIedge I acquired through you I also found out that disabling **autogain** altogether is of help too. 
At first I didn´t seem to be able to hear anything now but turning up "input" and "output" in the (single) **equalizer tab** 
to the max gives me acceptable sound/volume again.

Thanks again for your help. It´s highly appreciated.
Many greetings from Rosika
